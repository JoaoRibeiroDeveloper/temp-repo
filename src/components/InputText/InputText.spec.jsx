import { render, screen } from '@testing-library/react';
import { InputText } from '.';
import userEvent from '@testing-library/user-event';

describe('<InputText />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();

    render(
      <InputText
        handleChange={fn}
        searchValue={'testando'}
        placeholder={'teste'}
      />,
    );

    const input = screen.getByPlaceholderText(/teste/i);
    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(
      <InputText
        handleChange={fn}
        searchValue={'testando'}
        placeholder={'teste'}
      />,
    );

    const input = screen.getByPlaceholderText(/teste/i);
    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe('testando');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(
      <InputText
        handleChange={fn}
        searchValue={'testando'}
        placeholder={'teste'}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
