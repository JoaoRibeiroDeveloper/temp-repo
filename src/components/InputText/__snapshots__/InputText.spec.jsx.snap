// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText /> should match snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="teste"
    type="search"
    value="testando"
  />
</div>
`;
